network/modules/UDP/socket_wrapper.py

```diff
@@ -0,0 +1,137 @@
+import socket
+import struct 
+
+class UdpSocket:
+    """
+    Wrapper for Python's socket module.
+    """
+
+    def __init__(self, socket_instance: socket.socket = None) -> None:
+        """
+        Parameters
+        ----------
+        instance: socket.socket
+            For initializing Socket with an existing socket object.
+        """
+        if socket_instance is None:
+            self.__socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        else:
+            self.__socket = socket_instance
+
+    @classmethod
+    def send_to(self, data: bytes, host: str = "", port: int = 5000) -> bool:
+        """
+        Sends data to specified address
+
+
+        Parameters
+        ----------
+        data: bytes
+        host: str (default "")
+            Empty string is interpreted as '0.0.0.0' (IPv4) or '::' (IPv6), which is an open address
+        port: int (default 5000)
+            The host, combined with the port, will form the address as a tuple
+
+
+        Returns
+        -------
+        bool: if data was transferred successfully
+
+        """
+        address = (host, port)
+
+        data_sent = 0
+        data_size = len(data)
+        chunk_size = 4096
+
+        while data_sent + chunk_size < data_size:
```

Comment:
Don't put the check here, because if the user wanted to send 5000 bytes, the last 904 bytes will not be sent. Please keep this as `data_sent < data_size` and move the check to just prevent index out of bounds when (prevent it from trying to send 8192 bytes when there are only 5000 bytes)